indexprovider/wrapper.go

```diff
@@ -684,3 +711,38 @@ type basicDealInfo struct {
 	SectorID       abi.SectorID
 	DealProposal   storagemarket.ClientDealProposal
 }
+
+func (w *Wrapper) AnnounceBoostDirectDeal(ctx context.Context, entry *types.DirectDeal) (cid.Cid, error) {
+	// Filter out deals that should not be announced
+	if !entry.AnnounceToIPNI {
+		return cid.Undef, nil
+	}
+
+	contextID := []byte(entry.ID.String())
```

Comment:
Does this work? As we're parsing the context ID with uuid.FromBytes, I would expect that we would need to product the context ID with MarshallBinary()